app/workers/concerns/dump_worker.rb

```diff
@@ -40,16 +46,14 @@ def dump_type
     "project_#{dump_target}_export"
   end
 
-  def temp_file_path
-    "#{Rails.root}/tmp/#{project_file_path.join("_")}"
-  end
-
   def csv_file_path
-    "#{Rails.root}/tmp/#{project_file_path.join("_")}.csv"
+    @csv_tempfile ||= Tempfile.new(['export', '.csv'])
+    @csv_tempfile.path
   end
 
   def gzip_file_path
-    "#{Rails.root}/tmp/#{project_file_path.join("_")}.gz"
+    @gzip_tempfile ||= Tempfile.new(['export', '.gz'])
+    @gzip_tempfile.path
   end
 
   def project_file_path
```

Comment:
Yeah, using project ID sounds like a good idea.
> On 25 Feb 2016, at 10:05, Marten Veldthuis <notifications@github.com> wrote:
> 
> In app/workers/concerns/dump_worker.rb <https://github.com/zooniverse/Panoptes/pull/1687#discussion_r54072450>:
> 
> >    end
> >
> >    def gzip_file_path
> > -    "#{Rails.root}/tmp/#{project_file_path.join("_")}.gz"
> > +    @gzip_tempfile ||= Tempfile.new(['export', '.gz'])
> > +    @gzip_tempfile.path
> >    end
> >
> >    def project_file_path
> I'm actually contemplating getting rid of all of that, and just using the project id instead of logins and display names. Changing the replacement like you say means you could create "collisions" (not really collisions since the file name itself is a UUID, but you could get your file written in the same directory as another project which has a similar name but just difference in non-alphanum chars)
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/zooniverse/Panoptes/pull/1687/files#r54072450>.
>